### PR TITLE
Address security and linting for passport-azure-ad

### PR DESCRIPTION
**Pull request by jo-arroyo**
**NOTE: This PR replaces #3983, which had failing tests related to upgrading `mocha`.**

This PR adds linting and updates dependencies for the passport-azure-ad library. 

- Upgrades `grunt-contrib-nodeunit`
- Upgrades `https-proxy-agent`, addressing issue #3914 
- Upgrades `lodash` in the package-lock, addressing security advisory [1673](https://www.npmjs.com/advisories/1673)
- Upgrades `path-parse` in the package-lock, addressing security advisory [1773](https://www.npmjs.com/advisories/1773)
- Upgrades `diff` in the package-lock, addressing security advisory [1631](https://www.npmjs.com/advisories/1631)
- Adds `eslint`, `eslint-plugin-header`, `eslint-plugin-security` and lint rules in line with the rest of MSAL.js libraries. Lint errors and warnings addressed. 
- Cleans up comments.

Note: Does not upgrade `mocha`, despite low warning for prototype pollution(security advisory [1179](https://npmjs.com/advisories/1179)), as `mocha` is a dev dependency, and upgrading breaks existing tests.

### CODE REVIEW

File: maintenance/passport-azure-ad/lib/pem.js [hunk 1]
```diff
@@ -44,31 +25,33 @@
  */
 
 /* eslint-disable max-len */
-const CERT_REGEX = /-----BEGIN\sCERTIFICATE-----\s+([a-zA-Z0-9+\/=\r\n]+)\s+-----END\sCERTIFICATE-----/;
-const PRIVATE_KEY_REGEX = /-----BEGIN\sPRIVATE\sKEY-----\s+([a-zA-Z0-9+\/=\r\n]+)\s+-----END\sPRIVATE\sKEY-----/;
+const CERT_REGEX = /-----BEGIN\sCERTIFICATE-----\s+([a-zA-Z0-9+/=\r\n]+)\s+-----END\sCERTIFICATE-----/;
```
Comment (peterzenz) [inline maintenance/passport-azure-ad/lib/pem.js:28]: New regex is slightly different, is this correct?

File: maintenance/passport-azure-ad/lib/bearerstrategy.js [hunk 2]
```diff
@@ -163,7 +162,7 @@ const ttl = 1800; // 30 minutes cache
  */
 function Strategy(options, verifyFn) {
   passport.Strategy.call(this);
-  this.name = "oauth-bearer"; // Me, a name I call myself.
```
Comment (peterzenz) [inline maintenance/passport-azure-ad/lib/bearerstrategy.js]: :)

**Review by peterzenz — APPROVED** [review]

File: maintenance/passport-azure-ad/lib/pem.js (hunk 1 above)
Comment (jo-arroyo) [reply maintenance/passport-azure-ad/lib/pem.js]: Hmmm the linter was throwing a "no-useless-escape" error, but it looks like `/` should still be escaped, so I've added a rule disable for these few lines after checking the regex.

File: maintenance/passport-azure-ad/package.json [hunk 3]
```diff
@@ -46,7 +49,8 @@
     "valid-url": "^1.0.6"
   },
   "scripts": {
-    "test": "grunt run_tests"
+    "test": "grunt run_tests",
+    "lint": "./node_modules/.bin/eslint lib/**"
```
Comment (jasonnutter) [inline maintenance/passport-azure-ad/package.json:53]: I think this can just be

```suggestion
    "lint": "eslint lib/**"
```

**Review by jasonnutter — APPROVED** [review]

File: maintenance/passport-azure-ad/package.json (hunk 3 above)
Comment (jo-arroyo) [reply maintenance/passport-azure-ad/package.json:53]: After discussion, leaving the lint command as is. Changing to `eslint lib/**` results in `Invalid ecmaVersion` error.

File: maintenance/passport-azure-ad/lib/aadutils.js [hunk 4]
```diff
@@ -1,36 +1,19 @@
...
+const util = require("util");
 
-exports.getLibraryProduct = () => { return 'passport-azure-ad' };
+exports.getLibraryProduct = () => { return "passport-azure-ad" };
 exports.getLibraryVersionParameterName = () => { return "x-client-Ver" };
-exports.getLibraryProductParameterName = () => { return 'x-client-SKU' };
+exports.getLibraryProductParameterName = () => { return "x-client-SKU" };
 exports.getLibraryVersion = () => {
-  return '4.0.0';
+  return "4.0.0";
```
Comment (pkanher617) [inline maintenance/passport-azure-ad/lib/aadutils.js:16]: We should update the library version here, we are on version 4.3.0 now.

Comment (pkanher617) [inline maintenance/passport-azure-ad/lib/aadutils.js:16]:
```suggestion
  return "4.3.0";
```

**Review by pkanher617 — APPROVED** [review]
Great job, looks good other than one small thing for the version number

**Pull request merged**